semgrep/tests/performance/test_semgrep_rules_repo_perf.py

```diff
@@ -221,20 +221,27 @@
 
 
 @pytest.fixture(scope="session")
-def semgrep_rules_repo(tmp_path_factory):
+def semgrep_rules_repo(request, tmp_path_factory):
     repo_path = tmp_path_factory.mktemp("repo")
+
+    getoption = request.config.getoption
+    is_disabled = getoption("benchmark_skip") or getoption("benchmark_disable")
+    is_force_enabled = getoption("benchmark_enable") or getoption("benchmark_only")
+    if is_disabled and not is_force_enabled:
```

Comment:
Doesn't pytest have some existing functionality to do this for us? My concern is that we're duplicating internal pytest logic here, and that we could be missing an edge case they cover or that things could change out from underneath us and our logic no longer matches theirs. Thoughts?

Reply:
Yep, it is slightly scary. It's actually pytest-benchmark logic we're duplicating for the record.